### PR TITLE
Make DAML-LF 1.4 the default version output by damlc

### PR DESCRIPTION
**Pull request by hurryabit**
This removes the syntactic restriction on contract keys.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/1283)

### CODE REVIEW

File: docs/source/support/release-notes.rst [hunk 1]
```diff
@@ -9,6 +9,11 @@ This page contains release notes for the SDK.
 HEAD — ongoing
```
Comment (neil-da) [inline docs/source/support/release-notes.rst]: Release notes should say something about 1.4 since its something users end up seeing.

**Comment by remyhaemmerle-da** [comment]
Is this can wait for a week ? 
We would like to ship Enum types in daml-lf 1.4

File: docs/source/support/release-notes.rst (hunk 1 above)
Comment (hurryabit) [reply docs/source/support/release-notes.rst]: Added a not.

**Comment by hurryabit** [comment]
@remyhaemmerle-da Let's put enum types in DAML-LF 1.5 and release this now unless you have a strong argument against it.

**Pull request merged**